internal/mtail/examples_integration_test.go

```diff
@@ -92,7 +92,7 @@ func TestExamplePrograms(t *testing.T) {
 			waker, _ := waker.NewTest(ctx, 0) // oneshot means we should never need to wake the stream
 			store := metrics.NewStore()
 			programFile := filepath.Join("../..", tc.programfile)
-			mtail, err := mtail.New(ctx, store, mtail.ProgramPath(programFile), mtail.LogPathPatterns(tc.logfile), mtail.OneShot, mtail.OmitMetricSource, mtail.DumpAstTypes, mtail.DumpBytecode, mtail.LogPatternPollWaker(waker), mtail.LogstreamPollWaker(waker))
+			mtail, err := mtail.New(ctx, store, mtail.ProgramPath(programFile), mtail.LogPathPatterns(tc.logfile), mtail.OneShot, mtail.OmitMetricSource, mtail.DumpAstTypes, mtail.DumpBytecode, mtail.LogPatternPollWaker(waker), mtail.LogstreamPollWaker(waker), mtail.MaxRecursionDepth(100), mtail.MaxRegexLength(1024))
```

Comment:
If these options have defaults you shuoldnlt need to set them here.

Reply:
Unfortunately this is required for the unit tests pretty much across the board (ie. this file and etc).

The defaults are set in `cmd/main.go` via the `flag.Int` parameters (line 66 & 67 of that file in this merge request). In the production code, the `New` call on line 213 of `cmd/main.go` gets passed those flag values. Regardless whether it was set by the user or defaulted, the new call doens't know anything besides that it was passed a struct of `Option...` type. 

In this unit test, like the others, we have to take the role of passing those parameters. There is no `main.go` to do that for us since those tests don't have any interface to the command line (in the same way that main.go is a command line function). So hence we have all this duplication. 

I don't think this is the first time this has to be done. A similar example is the third argument here `mtail.ProgramPath(programFile)` is actually given a default on line 43 of `cmd/main.go` and passed into the new call on line 213 of that file. The unit tests don't invoke the main.go code so we have to duplicate that ProgramPath thing in a lot of tests.

This isn't great I agree, I'm happy to implement any suggestions to improve this if you have any. I'm not confident enough in this codebase to suggest one quite yet though!

Reply:
Never mind, I read your suggestion below. I agree I'll implement that.